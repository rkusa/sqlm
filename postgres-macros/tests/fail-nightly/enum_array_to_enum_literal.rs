use postgres_types::{FromSql, ToSql};
use sqlm_postgres::{sql, Enum};

#[derive(Debug, Default, FromSql, ToSql, Enum, PartialEq, Eq)]
#[postgres(name = "role")]
enum Role {
    #[default]
    #[postgres(name = "user")]
    User,
    #[postgres(name = "admin")]
    Admin,
}

#[tokio::main]
async fn main() {
    let _: Role = sql!(r#"SELECT ARRAY['admin','user']::role[]"#)
        .await
        .unwrap();
}
