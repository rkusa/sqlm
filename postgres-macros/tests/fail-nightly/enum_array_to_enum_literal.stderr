error[E0277]: the trait bound `Role: Query<sqlm_postgres::types::Array<Vec<sqlm_postgres::types::Enum<(EnumVariant<"admin">, EnumVariant<"user">)>>>>` is not satisfied
  --> tests/fail-nightly/enum_array_to_enum_literal.rs:17:10
   |
17 |         .await
   |         -^^^^^
   |         ||
   |         |the trait `Query<sqlm_postgres::types::Array<Vec<sqlm_postgres::types::Enum<(EnumVariant<"admin">, EnumVariant<"user">)>>>>` is not implemented for `Role`, which is required by `Sql<'_, sqlm_postgres::types::Array<Vec<sqlm_postgres::types::Enum<(EnumVariant<"admin">, EnumVariant<"user">)>>>, _>: IntoFuture`
   |         help: remove the `.await`
   |
   = help: the following other types implement trait `Query<Cols>`:
             <() as Query<()>>
             <Option<T> as Query<Primitive<<T as SqlType>::Type>>>
             <Option<T> as Query<Struct<Cols>>>
             <Option<Vec<u8>> as Query<Primitive<Bytea>>>
             <Vec<T> as Query<Primitive<<T as SqlType>::Type>>>
             <Vec<T> as Query<Struct<Cols>>>
             <Vec<T> as Query<sqlm_postgres::types::Array<Vec<<T as SqlType>::Type>>>>
             <Vec<Vec<u8>> as Query<Primitive<Bytea>>>
           and $N others
   = note: required for `Sql<'_, sqlm_postgres::types::Array<Vec<sqlm_postgres::types::Enum<(EnumVariant<"admin">, EnumVariant<"user">)>>>, Role>` to implement `IntoFuture`
